sdk/extensions/Microsoft.Extensions.Azure/tests/compatibility/ExpectedAotWarnings.txt

```diff
@@ -0,0 +1,11 @@
+AzureClientFactoryBuilder\.cs\(\d*\): Trim analysis warning IL2026: Microsoft\.Extensions\.Azure\.AzureClientFactoryBuilder\.\<\>c__DisplayClass\d*_\d*`2\.\<Azure\.Core\.Extensions\.IAzureClientFactoryBuilderWithConfiguration\<Microsoft\.Extensions\.Configuration\.IConfiguration\>\.RegisterClientFactory\>b__\d*\(!1,TokenCredential\): Using member 'Microsoft\.Extensions\.Azure\.ClientFactory\.CreateClient\(Type,Type,Object,IConfiguration,TokenCredential\)' which has 'RequiresUnreferencedCodeAttribute'
```

Comment:
The reason this is failing is because the warnings are showing up 2x in the following:

```
ILC : AOT analysis warning IL3050: Azure.Core.Pipeline.DiagnosticScope.ActivityAdapter.Dispose(): Using member 'System.Diagnostics.DiagnosticSource.Write(String,Object)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling. [D:\a\_work\1\s\eng\scripts\compatibility\TempAotCompatFiles\aotcompatibility.csproj] 
ILC : AOT analysis warning IL3050: Azure.Core.Pipeline.DiagnosticScope.ActivityAdapter.Dispose(): Using member 'System.Diagnostics.DiagnosticSource.Write(String,Object)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling. [D:\a\_work\1\s\eng\scripts\compatibility\TempAotCompatFiles\aotcompatibility.csproj] 
ILC : AOT analysis warning IL3050: Azure.Core.Pipeline.DiagnosticScope.ActivityAdapter.MarkFailed<T>(!!0,String): Using member 'System.Diagnostics.DiagnosticSource.Write(String,Object)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling. [D:\a\_work\1\s\eng\scripts\compatibility\TempAotCompatFiles\aotcompatibility.csproj] 
ILC : AOT analysis warning IL3050: Azure.Core.Pipeline.DiagnosticScope.ActivityAdapter.MarkFailed<T>(!!0,String): Using member 'System.Diagnostics.DiagnosticSource.Write(String,Object)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling. [D:\a\_work\1\s\eng\scripts\compatibility\TempAotCompatFiles\aotcompatibility.csproj] 
ILC : AOT analysis warning IL3050: Azure.Core.Pipeline.DiagnosticScope.ActivityAdapter.WriteStartEvent(): Using member 'System.Diagnostics.DiagnosticSource.Write(String,Object)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling. [D:\a\_work\1\s\eng\scripts\compatibility\TempAotCompatFiles\aotcompatibility.csproj] 
ILC : AOT analysis warning IL3050: Azure.Core.Pipeline.DiagnosticScope.ActivityAdapter.WriteStartEvent(): Using member 'System.Diagnostics.DiagnosticSource.Write(String,Object)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. DiagnosticSource may require creating new generic types or methods, which requires creating code at runtime. This may not work when AOT compiling. [D:\a\_work\1\s\eng\scripts\compatibility\TempAotCompatFiles\aotcompatibility.csproj] 
```

So I guess I will need to duplicate those lines in the ExpectedWarnings.txt file.

Reply:
Ok that makes sense. I had opted to enforce warning count to be exact to help prevent lists from getting stale.